nmoscommon/auth/claims_validator.py

```diff
@@ -32,19 +35,51 @@ def validate_aud(self):  # placeholder
 
     def validate_nmos(self):
         claim_name = "x-nmos-api"
-        valid_claim_option = self.options.get(claim_name)
-        valid_claim_value = valid_claim_option.get('value')
+        actual_claim_value = self.get(claim_name)  # actual claim value in JWT
+        valid_claim_option = self.options.get(claim_name)  # from `claim_options.py`
 
-        actual_claim_value = self.get(claim_name)
-        if not actual_claim_value:
+        if not actual_claim_value:  # Missing x-nmos-api claim
             raise MissingClaimError(claim_name)
-        if not valid_claim_value:
+        if not valid_claim_option:  # No options given for validation
             return
 
-        # TODO - add validation for versions and read/write
-        for attr in valid_claim_value:
-            if actual_claim_value.get(attr) != valid_claim_value.get(attr):
-                raise InvalidClaimError(claim_name)
+        def _validate_permissions_object(valid_api_value):
+            access_permission_object = actual_claim_value.get(valid_api_value)
+            if not access_permission_object:
+                raise InvalidClaimError("{}. No entry in claim for '{}'.".format(claim_name, valid_api_value))
+
+            if request.method in ["GET", "OPTIONS", "HEAD"]:
+                access_right = "read"
+            else:
+                access_right = "write"
+            url_access_list = access_permission_object.get(access_right)
+            if not url_access_list:
+                raise InvalidClaimError(
+                    "{}. No entry in permissions object for '{}'.".format(claim_name, access_right))
+
+            pattern = compile(r'/x-nmos/[a-zA-Z]+/v[0-9]\.[0-9]/(.*)')  # Capture path after namespace
+            trimmed_path = pattern.match(request.path).group(1)
+            for wildcard_url in url_access_list:
+                if fnmatch(trimmed_path, wildcard_url):
+                    return True
+
+            raise InvalidClaimError("{}. No matching paths in access list for '{}'.".format(claim_name, trimmed_path))
+
+        # Single value present in claims options
+        valid_api_value = valid_claim_option.get('value')
```

Comment:
It's not obvious to me what this is doing. Could you add further comments by way of example, or change the variable names if appropriate?